### PR TITLE
Fix bug that prevented regular users to spawn notebooks.

### LINKED ISSUE #835: [Bug]: Basic users cannot spawn jupyter notebook from Spawner page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Dashboard basic users cannot spawn jupyter notebook from the Spawner page due to lack of sufficient permissions.

Dashboard Admins are able to spawn.

The logs ([rhods-dashboard-5b5999f447-66rv6-rhods-dashboard.log](https://github.com/secure/attachment/12791337/12791337_rhods-dashboard-5b5999f447-66rv6-rhods-dashboard.log)) in dashboard pod report:

{"level":50,"time":1669921101154,"pid":108,"hostname":"rhods-dashboard-5b5999f447-66rv6","msg":"A
 Non-Admin User (ldap-user19) made a request against an endpoint that 
requires an admin."}

{"level":50,"time":1669921101385,"pid":108,"hostname":"rhods-dashboard-5b5999f447-66rv6","msg":"Failed
 to delete notebook, notebooks.kubeflow.org \"jupyter-nb-ldap-2duser19\"
 not found}"}

### Expected Behavior

Regular users should be able to spawn

### Steps To Reproduce

1. Log in RHODS Dashboard > Enabled as basic user A (from rhods-users group)
2. Launch Jupyter and try to spawn a notebook
3. check the outcome
4. Log in RHODS as RHODS Admin
5. repeat 2-3
6. As RHODS Admin go to Dashboard > Settings > User management 
7. add "rhods-users" among the RHODS administrators
8. repeat 2-3 using user A once again (now the user will be able to spawn)

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Fix #835 
* Omitted username field in spawner page for regular calls, as it should only be used when impersonating
* Revamped the impersonatedUsername field

## How Has This Been Tested?

1. Log in Dashboard > Enabled as regular user 
2. Launch Jupyter and try to spawn a notebook
3.Now it should be able to create a notebook

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/pages/notebookController/notebookControllerContextTypes.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export type NotebookControllerContextProps = {
    */
   setImpersonating: SetImpersonating;
   /** The user that is impersonated -- null is no impersonation */
-  impersonatedUsername: string | null;
+  impersonatedUsername: string | undefined;
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/notebookController/notebookControllerContextTypes.ts:24]:
```suggestion
  impersonatedUsername?: string;
```

**Review by andrewballantyne — APPROVED** [review]
/hold

LGTM, should get some on-cluster testing. @lucferbux do you have this hosted anywhere?

**Comment by lucferbux** [comment]
> /hold
> 
> LGTM, should get some on-cluster testing. @lucferbux do you have this hosted anywhere?

Yes, sending to you credentials in a private chat

**Comment by andrewballantyne** [comment]
/unhold
/approve

Sweet, works like a charm.

**Pull request merged**